### PR TITLE
Porting tf2_sensor_msgs in C++ (#2)

### PR DESCRIPTION
**Pull request by SteveMacenski**
@clalancette Lets try this again :100:

### CODE REVIEW

**Comment by clalancette** [comment]
Cool, here is another CI run:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5353/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2083/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4446/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5316/)

**Comment by SteveMacenski** [comment]
Mmh. I don't actually see where tf2_sensor_msgs is being built in that output. The issues appear to me to be upstream of this. Thoughts?

**Comment by clalancette** [comment]
> Mmh. I don't actually see where tf2_sensor_msgs is being built in that output. The issues appear to me to be upstream of this. Thoughts?

It's here: https://ci.ros2.org/job/ci_linux/5353/consoleFull#console-section-378 .  The reason you don't see "much" being done is because this looks like a header-only library in C++, so all it is doing is installing the header file and surrounding infrastructure.  The problems that are causing this to be yellow are indeed elsewhere in the codebase, so we can ignore those.  Assuming the other builds finish with the same set of errors, I'll be happy to approve.

**Review by clalancette — APPROVED** [review]
Yellow CI is because of other problems in the codebase unrelated to this PR.

**Comment by SteveMacenski** [comment]
Awesome, merge as you see fit

**Comment by clalancette** [comment]
Done, thanks for the contribution.

File: tf2_sensor_msgs/package.xml [hunk 1]
```diff
@@ -9,23 +9,20 @@
...
+  <buildtool_depend>ament_cmake_auto</buildtool_depend>
 
-  <run_depend>cmake_modules</run_depend>
-  <run_depend>eigen</run_depend>
-  <run_depend>sensor_msgs</run_depend>
-  <run_depend>tf2_ros</run_depend>
-  <run_depend>tf2</run_depend>
-  <run_depend>python_orocos_kdl</run_depend>
+  <depend>cmake_modules</depend>
+  <depend>Eigen3</depend>
```
Comment (ruffsl) [inline tf2_sensor_msgs/package.xml:16]: I'm not sure capital `Eigen3` is defined in rosdistro.
```
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
tf2_sensor_msgs: Cannot locate rosdep definition for [Eigen3]
```

@clalancette , was this migration intentional?

Comment (SteveMacenski) [reply tf2_sensor_msgs/package.xml:16]: @ruffsl I think you're right, I looked at rosdistro and it looks like `eigen` leads to `eigen3` debians

Comment (ruffsl) [reply tf2_sensor_msgs/package.xml:16]: See https://github.com/ros2/geometry2/pull/76

Comment (clalancette) [reply tf2_sensor_msgs/package.xml:16]: I actually don't remember at all, sorry.  However, this package *is* released into Bouncy, so I'm pretty sure what is currently in there works one way or another.

**Pull request merged**